heartbeat/monitors/active/http/http.go

```diff
@@ -82,7 +82,7 @@ func create(
 }
 
 func newRoundTripper(config *Config, tls *transport.TLSConfig) (*http.Transport, error) {
-	var proxy func(*http.Request) (*url.URL, error)
+	proxy := http.ProxyFromEnvironment
```

Comment:
To keep the current behaviour, I'm wondering if we should first check if `HTTP_PROXY` is set. Main problem is it would not support all env variables supported by `ProxyFromEnvironment`.

Reply:
This would mean also checking `HTTPS_PROXY` and both lowercase spellings to be compatible with the `ProxyFromEnvironment`. There is also the option of creating a dummy `Request` and checking whether a proxy should be used according to `ProxyFromEnvironment`.

Reply:
The HTTP module does not load proxy settings from the environment on purpose. For heartbeat there are potentially two networks. One being the network/service to be monitored and the other being the one we send the results too. The elasticsearch output uses ProxyFromEnvironment in order to support proxy setting when publishing events.

In heartbeat monitors one should rely on provided settings as much as possible (one can reference environment variables in config files btw.). I'm even considering to put a DNS client with configurable resolver and caching strategy into heartbeat.

Reply:
To add to this @kvalev, this is a great feature to add, but I think we need to handle DNS resolution in a more controlled manner. Are you interested in updating this PR to do that?

Reply:
Thanks for the feedback.

I can somewhat see how the first point that I raised (proxy support in heartbeat) can be considered a feature. And you are right that one can explicitly set `proxy_url` for a specific group of hosts, instead of having to rely on the underlying configuration. I was just thinking it might be a nice idea to have it natively in heartbeat. It is fair if you see it differently, but then the docs should be updated, as they are straight up incorrect (I am referring to the `proxy_url` documentation, which mentions that the `HTTP_PROXY` environment variable is taken into account).

However the second point that I raised (that I have provided unit tests tests for) seems to be a bug from what I can see. I was looking into fixing the issue myself when I initially stumbled upon it, but it seems to be a bit more complex for my newbie Go skills and inexperience with the provided HTTP API. I was hoping that it might be much easier to be fixed by one of the core developers, given the provided test cases.

Reply:
> I can somewhat see how the first point that I raised (proxy support in heartbeat) can be considered a feature. And you are right that one can explicitly set proxy_url for a specific group of hosts, instead of having to rely on the underlying configuration. I was just thinking it might be a nice idea to have it natively in heartbeat. It is fair if you see it differently, but then the docs should be updated, as they are straight up incorrect (I am referring to the proxy_url documentation, which mentions that the HTTP_PROXY environment variable is taken into account).

You are right about the doc being incorrect. Clearly a doc bug. Mostly likely some copy-n-paste error :(
I will submit a PR removing the mention of HTTP_PROXY in the docs.
Thank you for pointing this out!

> However the second point that I raised (that I have provided unit tests tests for) seems to be a bug from what I can see. I was looking into fixing the issue myself when I initially stumbled upon it, but it seems to be a bit more complex for my newbie Go skills and inexperience with the provided HTTP API. I was hoping that it might be much easier to be fixed by one of the core developers, given the provided test cases.

I didn't dig into the unit test, but yeah, sounds like a bug. Thanks for taking the time to write unit tests! This PR seems to solve two things at the same time. How about splitting it up, such that we can review and possibly merge the bug fix in a separate PR.

Reply:
See #7831 for documentation update.